bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/util/ByteStringer.java

```diff
@@ -54,6 +54,15 @@ public static ByteString wrap(final byte[] array) {
     return USE_ZEROCOPYBYTESTRING? ZeroCopyByteStringUtil.wrap(array): ByteString.copyFrom(array);
   }
 
+  /**
+   * Wraps a byte array in a {@link ByteString} without copying it.
+   */
+  public static ByteString wrap(final byte[] array, int offset, int length) {
+    return USE_ZEROCOPYBYTESTRING?
```

Comment:
When adapting a Result RowCell, `(offset == 0 && length == array.length)` will be true.

The 3 byte[] copies come into play for Puts that are sent via gRPC RPCs.  This case isn't an RPCE, but rather Dataflow's serialization process.  in the normal Dataflow serialization process, this will occur:

1. A scan happens, producing a Result generated from FlatRows.
2. Dataflow may serialize the Result 
3. if 2 occurs, deserialization occurs.

2 and 3 require byte array copies, regardless of whether the `Result` has `RowCells` or `KeyValue`s.  If the `Result` has `RowCell`s, there's no need to copy the `byte[]` as part of the serialization process.  If the `Result` has `KeyValue`s, then we the optimal solution would probably look different, and I don't think that case is something we should tackle in the Dataflow code, since it's not going to happen in production anyway.